Framework/Beamline/test/ComponentInfoTest.h

```diff
@@ -0,0 +1,78 @@
+#ifndef MANTID_BEAMLINE_COMPONENTINFOTEST_H_
+#define MANTID_BEAMLINE_COMPONENTINFOTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidBeamline/ComponentInfo.h"
+
+using Mantid::Beamline::ComponentInfo;
+
+class ComponentInfoTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static ComponentInfoTest *createSuite() { return new ComponentInfoTest(); }
+  static void destroySuite(ComponentInfoTest *suite) { delete suite; }
+
+  void test_empty_size() {
+    ComponentInfo info;
+    TS_ASSERT_EQUALS(info.size(), 0);
+  }
+
+  void test_empty_throw_out_of_range() {
+    ComponentInfo info;
+    TSM_ASSERT_THROWS(
+        "Empty ComponenetInfo, so any attempt to use this should throw",
+        info.detectorIndices(0), std::out_of_range &);
```

Comment:
I am not doing any range checks in `DetectorInfo` or `SpectrumInfo` (behaving basically like `std::vector::operator[]`). Should we keep that consistent?